### ISSUE TITLE
Rename keyboard IME and view

### ISSUE THREAD

**Issue by andrewtavis** [scribe-org/Scribe-Android#223]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/scribe-org/Scribe-Android/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow Scribe-Android's [Code of Conduct](https://github.com/scribe-org/Scribe-Android/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

There are two legacy names that should be changed in the project:

- `SimpleKeyboardIME` (from the original project)
- `MyKeyboardView` (should be named better)
- `MyKeyboard` (should be named better)

The question is, what should they be named to? `BaseKeyboardIME` and `BaseKeyboardView`?

### Contribution

Happy to discuss and review as needed 😊

**Comment by saurav0523** [comment]
can i start working on it

**Comment by saurav0523** [comment]
oe you need to assigned me

**Comment by andrewtavis** [comment]
Sounds good, @saurav0523! Thanks for the offer to help :) Let us know if you need some assistance here.

**Comment by andrewtavis** [comment]
Hi @saurav0523 👋 Checking in with you here. Is there anything we can do to support you with this?

**Comment by saurav0523** [comment]
Hi @andrewtavis thank you for assigned me.. as of now I'm bit busy for that just need some time.. Will start working on it after some days. Thank you

**Comment by andrewtavis** [comment]
All good, @saurav0523! Let us know if we can help :)

**Comment by andrewtavis** [comment]
Hey @saurav0523 👋 Checking in to see if there's anything that we can do to support here :) Please let us know if there's something we can do to help!

@angrezichatterbox: The renames I'm thinking of now would be:

- `SimpleKeyboardIME` -> `KeyboardIME`
- `MyKeyboardView` -> `KeyboardView`
- `MyKeyboard` -> `KeyboardBase`

**Comment by Linfye** [comment]
Maybe
`SimpleKeyboadIME` to `LanguageKeyboardIME` or `
GeneralKeyboardIME`?

The names you suggest all start with the word Keyboard. Are you doing this on purpose? What's the intention behind it? Like it'll be easier to be found or sorted.

**Comment by angrezichatterbox** [comment]
> @angrezichatterbox: The renames I'm thinking of now would be:
> 
> * `SimpleKeyboardIME` -> `KeyboardIME`
> * `MyKeyboardView` -> `KeyboardView`
> * `MyKeyboard` -> `KeyboardBase`
> 
> How do the above changes sound to you?

These sound nice. We could name the KeyboardIME as GeneralKeyboardIME as well as suggested by @Linfye 

Sorry for the delay. I missed this tag.

**Comment by andrewtavis** [comment]
I'm also good with `GeneralKeyvoardIME` :) @Linfye, would you have interest in making the changes?

**Comment by Linfye** [comment]
Certainly, but maybe in few days. I am busy with some college work these days. :)

**Comment by andrewtavis** [comment]
Sounds great! Good luck with the college work :) I'll assign you and you can start when you're ready 😊

**Comment by andrewtavis** [comment]
@saurav0523, you'd also be welcome to still work on this. Please let us know if you have time!

**Comment by saurav0523** [comment]
@andrewtavis heyy, I'm really sorry for the delay. I hope you understand this December month generally we are having our college end semester exams and all due to this Messed up. I'm not able to focus on this work but just give me few more weeks I'll give update on this. I really want to work on this issue

**Comment by andrewtavis** [comment]
Sounds good, @saurav0523! Let us know when you're able to start and we'll check in at the end of the month if we haven't heard from you 😊

@Linfye, let's hold up on this for now on your end :) Thanks for your willingness to help!

**Comment by Linfye** [comment]
OK :)

**Comment by saurav0523** [comment]
Sure

**Comment by akshayw1** [comment]
Is this completed?

**Comment by andrewtavis** [comment]
@saurav0523 is still assigned this, @akshayw1 :) 

@saurav0523, could you give a quick update on whether you'd be able to get to this in the next two weeks?

**Comment by andrewtavis** [comment]
Closed by #294 :) Thanks for picking this up and for the work in defining the solution heree, @Linfye! 🚀

@saurav0523: You'd be welcome to pick up another issue at some point, but we really needed to get this in so that we wouldn't be dealing with potential merge conflicts for the many changes to the keyboard on the way. Hope we can work with you again in the future! 😊 

@akshayw1: I'd suggest bringing these changes in for your PR for #267 as there could be conflicts given the new naming criteria :)
